tests/test_ModelsTest.py

```diff
@@ -52,3 +52,22 @@ def test_missing_page_cite_conversion(self):
         self.assertIsNone(citation1.groups["page"])
         self.assertIsNone(citation2.groups["page"])
         print("✓")
+
+    def test_corrected_reporter_hash(self):
+        """Does comparison_hash takes into account the CORRECTED reporter name"""
+        print("Testing corrected reporter in comparison_hash...", end=" ")
+        citation1 = case_citation(2, volume="2", reporter="U.S.", page="4")
+        citation2 = case_citation(2, volume="2", reporter="U. S.", page="4")
+        assert (
+            citation1.comparison_hash() == citation2.comparison_hash()
+        ), "Hashes should correct the name of the reporter"
```

Comment:
Sorry, this whole thing is super confusing. I fear I steered you wrong initially. I see now that the `comparison_hash()` method is used exclusively by the default `Resource` object (the thing a citation "points" to) to test whether two resources are equal or not. I believe this logic should be removed from the `CitationBase` and `ResourceCitation` classes, and returned to the `Resource` class itself for clarity. (I.e., basically roll back the changes in `models.py` made here: https://github.com/freelawproject/eyecite/commit/dfa53afae8ac68d7f81befc4dd0ee1645ace3c42#diff-e3def4308ba0fd02b8a71822c05c0cffe676777de880c85365d104314bc2f03d)

Next, I think we should remove the `@dataclass(eq=True, unsafe_hash=True)` decorators on the `CitationBase` class (and possibly subclasses, not sure how the inheritance will work) so that we can manually define a `CitationBase.__hash__()` and a `CitationBase.__eq__()` method. Right now, these methods are automatically generated from attributes defined using the `field()` function (so just the `groups` attribute), but here we should implement the custom logic you wrote where the `groups.reporter` key is compared differently (basically the logic in your current version of `comparison_hash()`).

I think/hope that if we define the top-level `__hash__()` function in this way, all our desired properties will follow through. E.g., you'll just be able to call `hash(cite)` and get the unique identifier (robust to corrected and uncorrected reporters) that you need.

Again, sorry that this has been so complicated! If you have a better way to proceed, feel free to say so.

Reply:
No worries at all. :+1: 
I will probably come back to implementing this feature next week because I am a bit busy for now